### PR TITLE
Add ability to specify SQLite synchronous mode and cache_size in the config. fix #817

### LINKED ISSUE #817: Heavy I/O with sqlite on Windows
When I use sqlite on windows PC my I/O starts to go up to 100% and the sync rate is very slow.
Seems the sqlite is by default in FULL synchronous mode.

I should be able to set PRAGMA synchronous = OFF as parameter for the session.

Please add this paramter to the config in the node. (like WAL)

Hint
On Ubuntu it seeems to work fine

### PR DESCRIPTION
**Pull request by Wekuz**

### CODE REVIEW

File: src/brs/props/Props.java [hunk 1]
```diff
@@ -122,6 +122,8 @@ public class Props {
   public static final Prop<Boolean> DB_H2_DEFRAG_ON_SHUTDOWN = new Prop<>("Db.H2.DefragOnShutdown", true);
   public static final Prop<Boolean> DB_OPTIMIZE = new Prop<>("DB.Optimize", true);
   public static final Prop<String> DB_SQLITE_JOURNAL_MODE = new Prop<>("DB.SqliteJournalMode", "WAL");
+  public static final Prop<String> DB_SQLITE_SYNCHRONOUS = new Prop<>("DB.SqliteSynchronous", "WAL");
```
Comment (frankTheTank72) [inline src/brs/props/Props.java:125]: SET as default "NORMAL"

File: src/brs/db/sql/dialects/DatabaseInstanceSqlite.java [hunk 2]
```diff
@@ -27,6 +27,19 @@ private String getJournalMode(){
     return "WAL";
   }
 
+  private String getSynchronousMode(){
```
Comment (frankTheTank72) [inline src/brs/db/sql/dialects/DatabaseInstanceSqlite.java:30]: Only allow OFF, NORMAL, FULL as value - set NORMAL as default if not set correct

**Review by frankTheTank72 — CHANGES_REQUESTED** [review]
Maybe i am also missing it - but how are the new paramters are used ?
@ohager  ?

**Comment by Wekuz** [comment]
@frankTheTank72 the value of the properties are retreived in the functions getSynchronousMode() and getCacheSize():
`String synchronous = propertyService.getString(Props.DB_SQLITE_SYNCHRONOUS).toUpperCase();`
and
`int cacheSize = propertyService.getInt(Props.DB_SQLITE_CACHE_SIZE);`
and is then used to set the properties in the DB session (the same way journal_mode is set) with the function:
```
config.addDataSourceProperty("synchronous", getSynchronousMode());
config.addDataSourceProperty("cache_size", getCacheSize());
```

**Comment by Wekuz** [comment]
I can prove it by running the node with a config that has these properties in it:
DB.SqliteSynchronous = OFF                                                                                                                                                                                                                   DB.SqliteCacheSize = -4000
Then I get this output:
![image](https://github.com/signum-network/signum-node/assets/89638089/014f2ade-5bc5-45fa-a59c-e7fd7cca26a0)
showing that the properties were set correctly. 

This custom output was gotten by adding these lines at the start of onStartupImpl():
```
Db.useDSLContext(ctx -> {
      logger.info(ctx.fetch("PRAGMA journal_mode").formatJSON());
      logger.info(ctx.fetch("PRAGMA synchronous").formatJSON());
      logger.info(ctx.fetch("PRAGMA cache_size").formatJSON());
    });
```

File: conf/mainnet/node.sqlite.properties [hunk 3]
```diff
@@ -14,6 +14,18 @@ DB.Optimize = on
 ## Info: MEMORY journal mode is not supported
 DB.SqliteJournalMode = WAL
 
+## Sqlite synchronous
+## https://www.sqlite.org/pragma.html#pragma_synchronous
+## Lowering this parameter can increase performance, but decrease reliability.
+## FULL (default) is the safest, but NORMAL should be also fine for running a local node.
+## Setting this to OFF could lead to database corruption on opertaing system crash or power loss!
+DB.SqliteSynchronous = FULL
```
Comment (frankTheTank72) [inline conf/mainnet/node.sqlite.properties:22]: set to:

# DB.SqliteSynchronous = FULL

**Review by frankTheTank72 — CHANGES_REQUESTED** [review]
Small change in the config

File: conf/mainnet/node.sqlite.properties (hunk 3 above)
Comment (Wekuz) [reply conf/mainnet/node.sqlite.properties:22]: Is it at least the right place to put it?

**Review by frankTheTank72 — APPROVED** [review]

**Review by ohager — APPROVED** [review]

**Pull request merged**